File: 1071 - Greatest Common Divisor of Strings.cpp
#include<string>

using namespace std;

class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        string s1 = str1 + str2, s2 = str2 + str1;
        if (s1 == s2) {
            return s1.substr(0, gcd(str1.size(), str2.size()));
        } else {
            return "";
        }
    }

    int gcd(int a, int b) {
        int r = a % b;
        while (r != 0) {
            a = b;
            b = r;
            r = a % b;
        }
        return b;
    }
};

File: 1071 - Greatest Common Divisor of Strings.py
import math

class Solution:

    def gcdOfStrings(self, str1: str, str2: str) -> str:
        g = math.gcd(len(str1), len(str2))
        res = str1[: g]
        if str1 + str2 == str2 + str1:
            return res
        else:
            return ''